cpu/stm32/periph/adc_f0_g0.c

```diff
@@ -86,7 +96,10 @@ int32_t adc_sample(adc_t line,  adc_res_t res)
 
     /* lock and power on the ADC device  */
     prep();
-
+    /* check if this is the VBAT line */
+    if (IS_USED(MODULE_PERIPH_VBAT) && line == VBAT_ADC) {
+        vbat_enable();
```

Comment:
I suppose this can't be done in `vbat_sample_mv()`?

Reply:
``` C
    vbat_enable();
    int32_t mv = adc_sample(VBAT_ADC, VBAT_ADC_RES);
    mv = (mv * VBAT_ADC_SCALE * vref_mv()) / VBAT_ADC_MAX;
    vbat_disable();
    return mv;
```
```
2022-02-20 10:11:24,139 # VBAT: 19[mV]
2022-02-20 10:11:25,140 # VBAT: 16[mV]
2022-02-20 10:11:26,142 # VBAT: 19[mV]
2022-02-20 10:11:27,143 # VBAT: 19[mV]
2022-02-20 10:11:28,144 # VBAT: 22[mV]
```

Results are bogus. Maybe it´s not so clever to do this before the ADC driver calls it´s `prep()` function.